php/libraries/TimePoint.class.inc

```diff
@@ -272,6 +272,27 @@ class TimePoint
             );
         }
 
+        $config =& \NDB_Config::singleton();
+        $visitLabelSettings = $config->getSetting('visitLabel');
+        $isValid            = false;
+        foreach (\Utility::toArray($visitLabelSettings) as $visitLabelSetting) {
+            if ($visitLabelSetting['@']['subprojectID'] == $subprojectID) {
+                $items = \Utility::toArray(
+                    $visitLabelSetting['labelSet']['item']
+                );
+                foreach ($items as $item) {
+                    if ($item['#'] == $visitLabel) {
+                        $isValid = true;
```

Comment:
maybe add a break statement(s) once you know it's valid

Reply:
good point

Reply:
@davidblader done